pyvortex/src/array.rs

```diff
@@ -186,6 +191,63 @@ impl PyArray {
             .and_then(|arr| Bound::new(py, PyArray { inner: arr }))
     }
 
+    /// Keep only a contiguous subset of elements.
+    ///
+    /// Parameters
+    /// ----------
+    /// start : :class:`int`
+    ///     The start index of the range to keep, inclusive.
+    ///
+    /// end : :class:`int`
+    ///     The end index, exclusive.
+    ///
+    /// Returns
+    /// -------
+    /// :class:`vortex.encoding.Array`
+    ///
+    /// Examples
+    /// --------
+    ///
+    /// Keep only the second through third elements:
+    ///
+    ///     >>> a = vortex.encoding.array(['a', 'b', 'c', 'd'])
+    ///     >>> a.slice(1, 3).to_arrow_array()
+    ///     <pyarrow.lib.StringArray object at ...>
+    ///     [
+    ///       "b",
+    ///       "c"
+    ///     ]
+    ///
+    /// Keep none of the elements:
+    ///
+    ///     >>> a = vortex.encoding.array(['a', 'b', 'c', 'd'])
+    ///     >>> a.slice(3, 3).to_arrow_array()
+    ///     <pyarrow.lib.StringArray object at ...>
+    ///     []
+    ///
+    /// Unlike Python, it is an error to slice outside the bounds of the array:
+    ///
+    ///     >>> a = vortex.encoding.array(['a', 'b', 'c', 'd'])
+    ///     >>> a.slice(2, 10).to_arrow_array()
+    ///     Traceback (most recent call last):
+    ///     ...
+    ///     ValueError: index 10 out of bounds from 0 to 4
+    ///
+    /// Or to slice with a negative value:
+    ///
+    ///     >>> a = vortex.encoding.array(['a', 'b', 'c', 'd'])
+    ///     >>> a.slice(-2, -1).to_arrow_array()
+    ///     Traceback (most recent call last):
+    ///     ...
+    ///     OverflowError: can't convert negative int to unsigned
+    ///
+    #[pyo3(signature = (start, end, *))]
+    fn slice(&self, start: usize, end: usize) -> PyResult<PyArray> {
```

Comment:
Can we also add `__getitem__`? For now we can only support numbers or slices

Reply:
Sure, I'll add __getitem__ and scalar_at in another PR.